### PR TITLE
Set the workflow instance ready state to running in failover

### LINKED ISSUE #15573: [Bug] [Master] Master failover will not handle ready_stop and ready_pause workflow instance correctly.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When master failover, if the workflow instance belong to the failover master is ready_stop or ready_pause, then the task instance will not be killed and the workflow instance will go into finished.

This is caused we didn't handle the ready state in failover.

### What you expected to happen

The workflow instance state is correct with task instance state.

### How to reproduce

1. Kill master 
2. Stop workflow instance
3. Restart master.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

close #15573

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by fuchanghai — COMMENTED** [review]
hi @ruanwenjun  If do this, the historical state added in the method ``` processServiceImpl#constructProcessInstance``` will be different from the actual one. Can I modify the command_type of Recover_Tolerance_fault_process in the method `` processServiceImpl#constructProcessInstance``` to solve this problem?

**Comment by ruanwenjun** [comment]
> hi @ruanwenjun If do this, the historical state added in the method ` processServiceImpl#constructProcessInstance` will be different from the actual one. Can I modify the command_type of Recover_Tolerance_fault_process in the method `` processServiceImpl#constructProcessInstance``` to solve this problem?

LGTM

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java [hunk 1]
```diff
@@ -793,6 +793,14 @@ private Boolean checkCmdParam(Command command, Map<String, String> cmdParam) {
                 break;
             case RECOVER_TOLERANCE_FAULT_PROCESS:
                 // recover tolerance fault process
+                // If the workflow instance is in ready state, we will change to running, this can avoid the workflow
+                // instance
+                // status is not correct with taskInsatnce status
+                if (processInstance.getState() == WorkflowExecutionStatus.READY_PAUSE
+                        || processInstance.getState() == WorkflowExecutionStatus.READY_STOP) {
+                    // todo: If we handle the ready state in WorkflowExecuteRunnable then we can remove below code
```
Comment (rickchengx) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java:801]: May I ask why we did not handle the ready state directly in WorkflowExecuteRunnable, this approach seems to be a workaround

Comment (ruanwenjun) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java:801]: This PR is only a quick bug fix, we need to handle the ready status in WorkflowExecuteRunnable when it is a failover workflow in the future.

**Review by rickchengx — APPROVED** [review]
LGTM

**Review by fuchanghai — APPROVED** [review]
LGTM

**Pull request merged**